### ISSUE TITLE
Wormhole paths stay when uninstalling a plugin that adds them.

### ISSUE THREAD

**Issue by Karirawri** [endless-sky/endless-sky#5885]
**Describe the bug**
uninstalling a plugin that creates a wormhole (like ursa polaris) leaves the game confused. the wormhole disappears as expected, but the game still considers it to create paths. paths that can't be followed without a wormhole

**To Reproduce**
1. Install Ursa Polaris, though i suspect any other plugin that adds a wormhole should work
2. Proceed through it until the wormhole is open
3. Uninstall
4. Watch as plotting a course that would go through the wormhole always ends in the system where the wormhole should be, forcing manual travel

**Expected behavior**
Wormhole pathing should be ignored even if the related mod's data is kept

**Screenshots**
![image](https://user-images.githubusercontent.com/31363287/115946499-f2798800-a497-11eb-893a-3f63e1bb7237.png)

**System (please complete the following information):**
 - OS: [e.g. Windows 10 Pro, version 20H2 build 19042.867]
 - Game Source: Steam Beta
 - Version: 0.9.13

**Link to save file**
[Karimi Renard.txt](https://github.com/endless-sky/endless-sky/files/6368979/Karimi.Renard.txt)

**Additional context**
Save file has data from other plugins, but they were uninstalled as well. They shouldn't affect the problem in any way.

**Comment by Amazinite** [comment]
Does Ursa Polaris create the wormhole at game start or are they added via events?

**Comment by tehhowch** [comment]
Removing plugins is still generally a no-no in terms of any given pilot, so IMO wonkiness is to be expected when doing so. Future plugin support could extend to disabling loading a pilot when any associated plugins are not detected.

No link is displayed because that is the default behavior for wormholes--the planet definition must specify a valid `description` node to be drawn. Wormholes without this, such as the one to the Pug galaxy, are still valid.

If a plugin event was used to define the planet that is used to create the wormhole, then the game has no recourse to not link these systems by invisible wormhole--all data it can see is perfectly valid. (This is because the contents of an event are stored in the player file. Once an event has happened to a player, it cannot be undone.)


Plugins can adopt a design pattern in which they do not ever define or modify the planet used to create wormholes via events, in order to avoid this effect. An event would then only insert the appropriately named stellar object into the requisite systems to create the wormhole.

**Comment by Karirawri** [comment]
I'm reporting this in light of the new functionality of keeping plugin data on a save for future reinstalls and to prevent crashes from plugin missions that aren't present. This sounds like something that could be fixed by it, perhaps by flagging events based on what file generates them and putting them in cold storage if the file that created it is no longer present.

**Comment by tehhowch** [comment]
I don't agree that _things that already happened to a pilot_ should conditionally happen--this is effectively gaslighting the player. To me it is a problem better solved before the player loads the save, by preventing the load entirely.

To tag each event's serialized changes is non-trivial and requires both storing where every DataNode was found at the DataNode level, and allowing each DataNode to possibly override the location of a parent node by means of a special token syntax. Without the last part, every serialized change from the pilot file is attributed to the pilot file rather than a plugin, and this issue would still persist.

An alternative to that node-level tagging approach is to store this information in the game's model of an "event" node and also change how the player remembers events that have happened. This would involve completely changing the event serialization data format, which would break returning to an earlier game version with that pilot (highly discouraged anyway).  This approach would not be able to address scenarios where a plugin modifies events found elsewhere, which is totally possible today.

---

In short, we can handle certain cases, but it's possible for plugin content to be written in a way that means there is no detectable issue (i.e. no actual bug) when a player removes the plugin, because removing a plugin is not supported. That the game now attempts to not shred your pilot file when you remove a plugin is not the same as the game supporting plugin removal for pilots that used it.

**Comment by tehhowch** [comment]
Anyhow, there were indeed a few misses in the "undefined reference handling" PR w.r.t wormholes, so I've addressed those in a9bacfb8b and eaab0a5a8.

With the provided pilot, pathing now works correctly: 
![image](https://user-images.githubusercontent.com/20871346/115967752-eb836180-a4f9-11eb-9cd9-a1de5559dc7b.png)
